packages/react/src/hooks/useStorageURL.ts

```diff
@@ -14,31 +14,35 @@ export interface UseStorageURLResult {
  */
 export const useStorageURL = (
   key: string,
-  options?: S3ProviderGetConfig
+  options?: S3ProviderGetConfig,
+  fallbackURL?: string
 ): UseStorageURLResult & { fetch: () => () => void } => {
   const [result, setResult] = React.useState<UseStorageURLResult>({
     isLoading: true,
   });
 
   // Used to prevent an infinite loop on useEffect, because `options`
   // will have a different reference on every render
-  const serializedOptions = JSON.stringify(options);
+  const optionsRef = React.useRef(options);
 
   const fetch = () => {
     setResult({ isLoading: true });
 
+    const options = optionsRef.current;
     const promise = Storage.get(key, options);
 
     // Attempt to fetch storage object url
     promise
       .then((url) => setResult({ url, isLoading: false }))
```

Comment:
Have we tested what happens if the key is not an image?

Reply:
An URL will always be generated from the API, but if it is not meant for an image, the `img` tag will not load it and trigger `onError`